### PR TITLE
Fixed hang when multiple processes open same DB

### LINKED ISSUE #1112: Error not thrown when multiple processes connect to same datadir
Identified in this comment: https://github.com/prysmaticlabs/prysm/issues/1017#issuecomment-447559851

We should visibly fail when two processes attempt to access the datadir concurrently. 
The database package should maintain a lock of some kind and throw a fatal error if attempting to access a datadir which is already locked by another process.

### PR DESCRIPTION
**Pull request by kushagrasharma**
Resolves #1112  

---

# Description

Previously, the database package would hang when two processes tried to access the same datadir simultaneously. Now, there is a one second timeout and an error returned if two processes access simultaneously.

### CODE REVIEW

File: beacon-chain/db/db.go [hunk 1]
```diff
@@ -46,8 +48,11 @@ func NewDB(dirPath string) (*BeaconDB, error) {
 		return nil, err
 	}
 	datafile := path.Join(dirPath, "beaconchain.db")
-	boltDB, err := bolt.Open(datafile, 0600, nil)
+	boltDB, err := bolt.Open(datafile, 0600, &bolt.Options{Timeout: 1 * time.Second})
 	if err != nil {
+		if strings.Contains(err.Error(), "timeout") {
```
Comment (prestonvanloon) [inline beacon-chain/db/db.go]: Please change this to 
```go
if err == bolt.ErrTimeout {
```

File: beacon-chain/db/db.go [hunk 2]
```diff
@@ -46,8 +48,11 @@ func NewDB(dirPath string) (*BeaconDB, error) {
...
+			return nil, errors.New("DB locked by another process")
```
Comment (prestonvanloon) [inline beacon-chain/db/db.go]: Please change this to: `cannot obtain database lock, database may be in use by another process.`

**Comment by prestonvanloon** [comment]
Oh, my bad! You need to remove the punctuation at the end of your error message. Thanks!

**Comment by kushagrasharma** [comment]
No worries, fixed!

**Comment by prestonvanloon** [comment]
Hey @kushagrasharma, what happened here? This PR was ready to merge

**Comment by kushagrasharma** [comment]
Sorry! Accidentally closed this. Reopening now.

**Comment by prestonvanloon** [comment]
Yay thanks @kushagrasharma

**Pull request merged**